src/App.tsx

```diff
@@ -1,56 +1,91 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { DropDownMenu } from './components/DropDownMenu/DropDownMenu';
+
+const debounce = (
+  f: React.Dispatch<React.SetStateAction<string>>,
+  delay: number,
+) => {
+  let timerId: number;
+
+  return (...args: string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPersonSlug, setSelectedPersonSlug] = useState('');
+  const [isSugActive, setIsSugActive] = useState(false);
+
+  const visiblePeople = useMemo(() => {
+    const normalizedQuery = appliedQuery.toLowerCase().trim();
+
+    if (appliedQuery) {
+      setIsSugActive(true);
+    }
+
+    if (!appliedQuery) {
+      setSelectedPersonSlug('');
+      setIsSugActive(false);
+    }
+
+    return peopleFromServer.filter(person => (
+      person.name.toLowerCase().includes(normalizedQuery)
+    ));
+  }, [appliedQuery]);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const selectedPerson = visiblePeople.find(person => (
```

Comment:
Would be good to use useMemo() here to memorize the selectedPerson value to avoid performing the expensive find() operation repeatedly if the dependencies haven't changed.